acceptance/gossip_peerings_test.go

```diff
@@ -22,35 +22,56 @@ package acceptance
 import (
 	"fmt"
 	"math/rand"
-	"os"
 	"strings"
 	"testing"
 	"time"
 
 	"github.com/cockroachdb/cockroach/acceptance/localcluster"
+	"github.com/cockroachdb/cockroach/gossip"
 	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/log"
 )
 
-func checkGossipNodes(node *localcluster.Container) int {
-	var m map[string]interface{}
-	if err := node.GetJSON("", "/_status/gossip", &m); err != nil {
-		return 0
-	}
-	count := 0
-	infos := m["infos"].(map[string]interface{})
-	for k := range infos {
-		if strings.HasPrefix(k, "node:") {
-			count++
+type checkGossipFunc func(map[string]interface{}) error
+
+// hasPeers returns a checkGossipFunction that passes when the given
+// number of peers are connected via gossip.
+func hasPeers(expected int) checkGossipFunc {
+	return func(infos map[string]interface{}) error {
+		count := 0
+		for k := range infos {
+			if strings.HasPrefix(k, "node:") {
+				count++
+			}
+		}
+		if count != expected {
+			return util.Errorf("expected %d peers, found %d", expected, count)
 		}
+		return nil
+	}
+}
+
+// hasSentinel is a checkGossipFunc that passes when the sentinel gossip is present.
+func hasSentinel(infos map[string]interface{}) error {
+	if _, ok := infos[gossip.KeySentinel]; !ok {
+		return util.Errorf("sentinel not found")
 	}
-	return count
+	return nil
 }
 
-func checkGossipPeerings(t *testing.T, l *localcluster.Cluster, d time.Duration) {
-	expected := len(l.Nodes) * len(l.Nodes)
-	log.Infof("waiting for complete gossip network of %d peerings", expected)
+// hasClusterID is a checkGossipFunc that passes when the cluster ID gossip is present.
+func hasClusterID(infos map[string]interface{}) error {
+	if _, ok := infos[gossip.KeyClusterID]; !ok {
+		return util.Errorf("cluster ID not found")
+	}
+	return nil
+}
 
+// checkGossip fetches the gossip infoStore from each node and invokes the given
+// function. The test passes if the function returns 0 for every node,
+// retrying for up to the given duration.
+func checkGossip(t *testing.T, l *localcluster.Cluster, d time.Duration,
```

Comment:
maybe move this up near the declaration of `checkGossipFunc` so that the comment is near that type